tests/integration/sync_test.go

```diff
@@ -303,6 +310,10 @@ func Test_Sync(t *testing.T) {
 			teardown := setup(t, tc.initialKongFile)
 			defer teardown(t)
 
+			if kongVersion < "2.0" && len(tc.expectedState.Plugins) > 0 {
```

Comment:
Please consult the go-kong codebase for how to detect kong versions and have a function to skip such tests, we may need to do such things more frequently.

Reply:
I also recommend isolating tests that are version-specific rather than sibling assertion of a specific "state" for a specific version.
This additional complexity can get complicated as we add more tests.

An alternate method here will be to create a dedicated TestFoo that is disabled for version x or below, or to have "DisableFilter string" (that takes in a semver filter like >2.3.0) in the test table.

Reply:
We should export the go-kong functions for this and reuse them as much as possible; no reason to maintain separate versions of them since any Go project is likely going to include go-kong for other reasons anyway.

I've opened https://github.com/Kong/go-kong/pull/129 as well since the current Enterprise version code doesn't work properly for initial minor releases (they're turned into pre-releases so semver comparisons say they're older).

Reply:
@rainest  Added an issue on go-kong for that https://github.com/Kong/go-kong/issues/130